pyparsing/core.py

```diff
@@ -4511,44 +4515,40 @@ def recurse(self) -> List[ParserElement]:
         return [self.expr] if self.expr is not None else []
 
     def parseImpl(self, instring, loc, doActions=True):
-        if self.expr is not None:
-            try:
-                return self.expr._parse(instring, loc, doActions, callPreParse=False)
-            except ParseBaseException as pbe:
-                if not isinstance(self, Forward) or self.customName is not None:
-                    pbe.msg = self.errmsg
-                raise
-        else:
+        if self.expr is None:
             raise ParseException(instring, loc, "No expression defined", self)
 
+        try:
+            return self.expr._parse(instring, loc, doActions, callPreParse=False)
+        except ParseBaseException as pbe:
+            if not isinstance(self, Forward) or self.customName is not None:
+                pbe.msg = self.errmsg
+            raise
+
     def leave_whitespace(self, recursive: bool = True) -> ParserElement:
         super().leave_whitespace(recursive)
 
-        if recursive:
```

Comment:
Rejected. I feel that explicit separation of these conditions is outweighs the benefits of denesting this short bit of code.